source/MapDetailPanel.cpp

```diff
@@ -573,13 +573,32 @@ void MapDetailPanel::DrawKey()
 		// four largest visible governments are labeled in the legend.
 		vector<pair<double, const Government *>> distances;
 		for(const auto &it : closeGovernments)
+		{
+			if(!it.first)
+				continue;
 			distances.emplace_back(it.second, it.first);
+		}
 		sort(distances.begin(), distances.end());
-		for(unsigned i = 0; i < 4 && i < distances.size(); ++i)
+		int drawn = 0;
+		vector<pair<const string, const Color *>> alreadyDisplayed;
+		for(const auto &it : distances)
 		{
-			RingShader::Draw(pos, OUTER, INNER, GovernmentColor(distances[i].second));
-			font.Draw(distances[i].second->GetName(), pos + textOff, dim);
+			const string &displayName = it.second->GetName();
+			const Color &displayColor = it.second->GetColor();
+			auto foundIt = find_if(alreadyDisplayed.begin(), alreadyDisplayed.end(),
+					[&displayName, &displayColor](const pair<const string, const Color *> &item)
+					{
+						return (item.first == displayName) && (*item.second == displayColor);
+					});
```

Comment:
You can replace this with `std::make_pair(...)` and change the `find_if` into a simple `find`, if you change the vector to have a Color object (and not a pointer).

Reply:
I don't want to deal with copy constructors. Or, can I just put the reference from `Government::GetColor()` in there and trust it won't try to implicitly copy or something?

Reply:
> I don't want to deal with copy constructors

what do you mean.

> can I just put the reference from Government::GetColor() in there and trust it won't try to implicitly copy or something?

I'm pretty sure that doesn't work

Reply:
It's going to want to construct a new Color object, isn't it? `Color` doesn't seem to have a copy constructor defined. Is it going to take the values in the array across properly?

Reply:
Yeah. The default copy constructor is (fortunately lol) correct.